server/middlewares/rateLimiter.ts

```diff
@@ -0,0 +1,63 @@
+import { Context, Next } from "koa";
+import { defaults } from "lodash";
+import RateLimiter from "@server/RateLimiter";
+import env from "@server/env";
+import { RateLimitExceededError } from "@server/errors";
+import Logger from "@server/logging/Logger";
+import Redis from "@server/redis";
+import { RateLimiterConfig } from "@server/types";
+
+export function rateLimiter() {
+  return async function rateLimiterMiddleware(ctx: Context, next: Next) {
+    if (!env.RATE_LIMITER_ENABLED) {
+      return next();
+    }
+
+    const limiter = RateLimiter.getRateLimiter(ctx.path);
+
+    try {
+      await limiter.consume(`${ctx.path}:${ctx.ip}`);
```

Comment:
Wondering if using this as a key for the default rate limiter is a good idea :thinking: 

This will create keys corresponding to the default rate limiter in the format `rl:<path>:<ip>`. Although, having `path` makes sense for route specific rate limiters, it seems unnecessary for the default one. For the default one, `rl:<ip>` should suffice. Besides, we would reclaim some space too. Let me know your thoughts @tommoor .

Reply:
Modified as per [`5d11311`](https://github.com/outline/outline/pull/3857/commits/5d1131126e789ccf422f6e862bb45fa6ee09f5cc).